client/landing/jetpack-cloud/components/scan-threats/index.tsx

```diff
@@ -54,18 +58,19 @@ const ScanThreats = ( { site, threats, userHasCredentials }: Props ) => {
 	}, [] );
 
 	const confirmAction = React.useCallback( () => {
-		window.alert(
-			`We are going to ${ actionToPerform } threat ${ selectedThreat?.id } on site ${ site.name }`
-		);
+		const actionCreator = actionToPerform === 'fix' ? fixThreatAlert : ignoreThreatAlert;
 		closeDialog();
 		setFixingThreats( stateThreats => [ ...stateThreats, selectedThreat ] );
-	}, [ actionToPerform, closeDialog, selectedThreat, site ] );
+		dispatch( actionCreator( site.ID, selectedThreat.id ) );
+	}, [ actionToPerform, closeDialog, dispatch, selectedThreat, site ] );
 
 	const confirmFixAllThreats = React.useCallback( () => {
-		window.alert( `Starting to fix ${ threats.length } threats found...` );
+		threats.forEach( threat => {
```

Comment:
We should investigate this a bit more. Instead of performing them in a loop we might have to wait for a response from the api before we request to make another fix. I think for now this is fine.